sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -51,16 +51,31 @@ import org.apache.spark.util.collection.OpenHashMap
       _FUNC_(col, array(percentage1 [, percentage2]...)) - Returns the exact percentile value array
       of numeric column `col` at the given percentage(s). Each value of the percentage array must
       be between 0.0 and 1.0.
-    """)
+
+      _FUNC_(col, frequency, percentage) - Returns the exact percentile value of numeric
+      column `col` with frequency column `frequency` at the given percentage. The value of
+      percentage must be between 0.0 and 1.0.
+
+      _FUNC_(col, frequency, array(percentage1 [, percentage2]...)) - Returns the exact percentile
+      value array of numeric column `col` with frequency column `frequency` at the given
+      percentage(s).Each value of the percentage array must be between 0.0 and 1.0.
+
+      """)
 case class Percentile(
     child: Expression,
+    frequency : Expression,
     percentageExpression: Expression,
+    withFrqExpr : Boolean,
```

Comment:
Please see my comment below why we need to make a distinction either using flag or Option

I am inclined towards using a flag because switching to option would change the code in update
from
val frqValue = frequency.eval(input)
to 
val frqValue = frequency.getOrElse( unit).eval(input) 

But i think Option[Expression] would be better logically
Once we have an agreement if we need to have a distinction or not i will make the changes accordingly